source/extensions/filters/network/rocketmq_proxy/config.cc

```diff
@@ -0,0 +1,68 @@
+#include "extensions/filters/network/rocketmq_proxy/config.h"
+
+#include <cstdlib>
+
+#include "envoy/config/filter/network/rocketmq_proxy/v2alpha1/rocketmq_proxy.pb.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "extensions/filters/network/rocketmq_proxy/conn_manager.h"
+#include "extensions/filters/network/rocketmq_proxy/stats.h"
+#include "extensions/filters/network/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RocketmqProxy {
+
+namespace rocketmq_config = envoy::config::filter::network::rocketmq_proxy::v2alpha1;
+
+Network::FilterFactoryCb RocketmqProxyFilterConfigFactory::createFilterFactoryFromProtoTyped(
+    const rocketmq_config::RocketmqProxy& proto_config,
+    Server::Configuration::FactoryContext& context) {
+  std::shared_ptr<ConfigImpl> filter_config = std::make_shared<ConfigImpl>(proto_config, context);
+  return [filter_config, &context](Network::FilterManager& filter_manager) -> void {
+    filter_manager.addReadFilter(
+        std::make_shared<ConnectionManager>(*filter_config, context.dispatcher().timeSource()));
+  };
+}
+
+REGISTER_FACTORY(RocketmqProxyFilterConfigFactory,
+                 Server::Configuration::NamedNetworkFilterConfigFactory);
+
+const char* ConfigImpl::ENV_APP_VERSION = "ENVOY_APP_VERSION";
+
+ConfigImpl::ConfigImpl(const RocketmqProxyConfig& config,
+                       Server::Configuration::FactoryContext& context)
+    : context_(context), stats_prefix_(fmt::format("rocketmq.{}.", config.stat_prefix())),
+      stats_(RocketmqFilterStats::generateStats(stats_prefix_, context_.scope())),
+      develop_mode_(config.develop_mode()) {
+  char* appVersionPtr = std::getenv(ENV_APP_VERSION);
+  if (appVersionPtr != nullptr) {
+    version_ = std::string(appVersionPtr, strlen(appVersionPtr));
+    ENVOY_LOG(info, "App version specified is: {}", appVersionPtr);
+  } else {
+    ENVOY_LOG(info, "No app version specified");
+  }
+}
+
+std::string ConfigImpl::proxyAddress() {
+  const LocalInfo::LocalInfo& localInfo = context_.getServerFactoryContext().localInfo();
+  Network::Address::InstanceConstSharedPtr address = localInfo.address();
+  if (address->type() == Network::Address::Type::Ip) {
+    const std::string& ip = address->ip()->addressAsString();
+    std::string proxyAddr{ip};
```

Comment:
Generally, this is a development phase facility. 
RocketMQ SDK has embedded service discovery. At first step, it queries route info of the topic in interest, which contains the IP address and port of brokers.  Then it interacts with brokers.

In the production environment, we usually take advantage of the traffic intercepting instrument and feed all related traffics to the envoy. Whilst in the development phase, it would be nice to instruct SDK to point to envoy endpoint and everything works. One thing we need to fix is that the route info still points to actual remote hosts, instead of envoy itself. To rectify this issue, we offer settings to facilitate the development.